### PR TITLE
Render descriptions in flytectl

### PR DESCRIPTION
**Pull request by mayitbeegh**
# TL;DR
- Display input output variable description in the table view of `get task` `get workflow` `get launchplan`.
- Display input output variable description as comment when generating an execution spec template.

`get launchplan`
Screen Shot 2021-08-02 at 11 17 58

`get workflow`
Screen Shot 2021-08-02 at 11 18 18

`get task`
Screen Shot 2021-08-02 at 11 21 13

`get task --exexFile exec.yaml` before vs after
Screen Shot 2021-07-29 at 16 44 15

## Type
- [ ] Bug Fix
- [x] Feature
- [ ] Plugin

## Are all requirements met?

- [x] Code completed
- [x] Smoke tested
- [x] Unit tests added
- [x] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1281

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/printer/printer.go [hunk 1]
```diff
@@ -164,6 +170,69 @@ func printJSONYaml(format OutputFormat, v interface{}) error {
...
+func FormatVariableDescriptions(variableMap map[string]*core.Variable) {
+	keys := make([]string, 0, len(variableMap))
+	// sort the keys for testing and consistency with other output formats
+	for k := range variableMap {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	var descriptions []string
+	//for k, v := range variableMap {
```
Comment (katrogan) [inline pkg/printer/printer.go:182]: delete?

File: pkg/printer/printer.go [hunk 2]
```diff
@@ -164,6 +170,69 @@ func printJSONYaml(format OutputFormat, v interface{}) error {
...
+			descriptions = append(descriptions, getTruncatedLine(fmt.Sprintf("%s: %s", k, v.Description)))
+		} else {
+			descriptions = append(descriptions, getTruncatedLine(k))
+		}
+
+	}
+	variableMap[DefaultFormattedDescriptionsKey] = &core.Variable{Description: strings.Join(descriptions, "\n")}
+}
+
+func FormatParameterDescriptions(variableMap map[string]*core.Parameter) {
```
Comment (katrogan) [inline pkg/printer/printer.go:199]:
```suggestion
func FormatParameterDescriptions(parameterMap map[string]*core.Parameter) {
```

File: pkg/printer/printer.go [hunk 3]
```diff
@@ -164,6 +170,69 @@ func printJSONYaml(format OutputFormat, v interface{}) error {
...
+	// sort the keys for testing and consistency with other output formats
+	for k := range variableMap {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	var descriptions []string
+	//for k, v := range variableMap {
+	for _, k := range keys {
+		if k == DefaultFormattedDescriptionsKey {
```
Comment (katrogan) [inline pkg/printer/printer.go:184]: why is this check necessary? shouldn't the variable map just be straight from the entity closure?

File: pkg/printer/printer.go [hunk 4]
```diff
@@ -164,6 +170,69 @@ func printJSONYaml(format OutputFormat, v interface{}) error {
...
+			descriptions = append(descriptions, getTruncatedLine(fmt.Sprintf("%s: %s", k, v.Var.Description)))
+		} else {
+			descriptions = append(descriptions, getTruncatedLine(k))
+		}
+	}
+	variableMap[DefaultFormattedDescriptionsKey] = &core.Parameter{Var: &core.Variable{Description: strings.Join(descriptions, "\n")}}
+}
+
+func getTruncatedLine(line string) string {
+	// maybe add width to function signature later
```
Comment (katrogan) [inline pkg/printer/printer.go:228]:
```suggestion
	// TODO: maybe add width to function signature later
```

**Review by katrogan — APPROVED** [review]
lgtm! will let @pmahindrakar-oss review too

File: pkg/printer/printer.go (hunk 1 above)
Comment (mayitbeegh) [reply pkg/printer/printer.go:182]: oops

File: pkg/printer/printer.go (hunk 3 above)
Comment (mayitbeegh) [reply pkg/printer/printer.go:184]: There're some side effects in the tests that I don't fully understand. For example, if I change the order of the tests, or if I try to run tests individually, things break. This is one of the problems I ran into, the function under test got called twice.

File: cmd/get/execution_util.go [hunk 5]
```diff
@@ -6,25 +6,25 @@ import (
...
 	"os"
 
+	"gopkg.in/yaml.v3"
+
 	cmdUtil "github.com/flyteorg/flytectl/pkg/commandutils"
 	"github.com/flyteorg/flyteidl/clients/go/coreutils"
 	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
 	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
-
-	"sigs.k8s.io/yaml"
```
Comment (pmahindrakar-oss) [inline cmd/get/execution_util.go:14]: For consistency we should probably convert other cli structs to also use the new gopkg.
Can you create an issue for it .

File: pkg/printer/printer.go (hunk 3 above)
Comment (pmahindrakar-oss) [reply pkg/printer/printer.go:183]: @mayitbeegh If any test execution changes based on the order then that would be case if you are using global state across all the tests.
In such case move common state changes in the setup function
and override behavior for test specific states after the setup is called in the test. That should fix such cases.
We have done this across most of the tests in flytectl.
Also is it possible for you to just write unit tests covering this function, the bot is showing that this function is not covered anywhere

**Review by pmahindrakar-oss — APPROVED** [review]

File: pkg/printer/printer.go (hunk 3 above)
Comment (mayitbeegh) [reply pkg/printer/printer.go:183]: @pmahindrakar-oss Okay, I can fix that

**Review by pmahindrakar-oss — APPROVED** [review]
